p2p/simulations/adapters/exec.go

```diff
@@ -338,6 +338,31 @@ type execNodeConfig struct {
 	PeerAddrs map[string]string `json:"peer_addrs,omitempty"`
 }
 
+// ExternalIP gets an external IP address so that Enode URL is usable
+func ExternalIP() net.IP {
+	//addrs, err := net.InterfaceAddrs()
+	//if err != nil {
+	//log.Crit("error getting IP address", "err", err)
+	//}
+	//for _, addr := range addrs {
+	//if ip, ok := addr.(*net.IPNet); ok && !ip.IP.IsLoopback() {
+	//return ip.IP
+	//}
+	//}
```

Comment:
Did this commented code not work?

Reply:
On OSX (both locally and on Travis), this is returning an IPv6 address. In the previous version of the exec and docker adapters, this was not called, because we had 127.0.0.1 hardcoded.

I will review how to get an IPv4 address and probably revert this back, but for now I am leaving the hack-code which makes a UDP dial and gets the local address from there.

Reply:
@lmars I reverted your code, but also ignored the fe80::1 link-local address.